src/backend/distributed/utils/aggregate_utils.c

```diff
@@ -918,7 +918,7 @@ TypecheckWorkerPartialAggArgType(FunctionCallInfo fcinfo, StypeBox *box)
 										true, 'i', &argtypesNull);
 			Assert(!argtypesNull);
 			TupleDesc tupleDesc = box->aggregationArgumentContext->tupleDesc;
-			if (argType != tupleDesc->attrs[aggregateArgIndex].atttypid)
+			if (argType != TupleDescAttr(tupleDesc, aggregateArgIndex)->atttypid)
```

Comment:
I couldn't understand what is the difference between this, and the `Attr` function you defined in `columnar_version_compat.h`.

Reply:
I believe this answers it:
https://github.com/citusdata/citus/pull/7981/files#r2194873139